x/dclupgrade/client/cli/query_approved_upgrade_test.go

```diff
@@ -0,0 +1,164 @@
+package cli_test
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
+	"github.com/stretchr/testify/require"
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/network"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/nullify"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/dclupgrade/client/cli"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/dclupgrade/types"
+)
+
+// Prevent strconv unused error.
+var _ = strconv.IntSize
+
+func networkWithApprovedUpgradeObjects(t *testing.T, n int) (*network.Network, []types.ApprovedUpgrade) {
+	t.Helper()
+	cfg := network.DefaultConfig()
+	state := types.GenesisState{}
+	require.NoError(t, cfg.Codec.UnmarshalJSON(cfg.GenesisState[types.ModuleName], &state))
+
+	for i := 0; i < n; i++ {
+		approvedUpgrade := types.ApprovedUpgrade{
+			Plan: types.Plan{
+				Name: strconv.Itoa(i),
+			},
+		}
+		nullify.Fill(&approvedUpgrade)
+		state.ApprovedUpgradeList = append(state.ApprovedUpgradeList, approvedUpgrade)
+	}
+	buf, err := cfg.Codec.MarshalJSON(&state)
+	require.NoError(t, err)
+	cfg.GenesisState[types.ModuleName] = buf
+	return network.New(t, cfg), state.ApprovedUpgradeList
+}
+
+func TestShowApprovedUpgrade(t *testing.T) {
+	net, objs := networkWithApprovedUpgradeObjects(t, 2)
+
+	ctx := net.Validators[0].ClientCtx
+	common := []string{
+		fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+	}
+	for _, tc := range []struct {
+		desc   string
+		idName string
+
+		args []string
+		err  error
+		obj  types.ApprovedUpgrade
+	}{
+		{
+			desc:   "found",
+			idName: objs[0].Plan.Name,
+
+			args: common,
+			obj:  objs[0],
+		},
+		{
+			desc:   "not found",
+			idName: strconv.Itoa(100000),
+
+			args: common,
+			err:  status.Error(codes.InvalidArgument, "not found"),
+		},
+	} {
+		tc := tc
+		t.Run(tc.desc, func(t *testing.T) {
+			args := []string{
+				tc.idName,
+			}
+			args = append(args, tc.args...)
+			out, err := clitestutil.ExecTestCLICmd(ctx, cli.CmdShowApprovedUpgrade(), args)
+			if tc.err != nil {
```

Comment:
Please note that Starport generates an incorrect code for `not found` test case in single query CLI command tests.

Reply:
fixed